### PR TITLE
Added Generic lists and details for books and authors

### PR DESCRIPTION
**Pull request by RizwanNiazFolio3**
### Current Status
Currently the website does not have any functionality besides the home page. Clicking on the other links does nothing.

### Changes/Additions
This PR adds the functionality for views for the list of books as well as the list of authors.
Booklist
AuthorList

Clicking on any of these leads us to the details pages for these classes. i.e the details of a particular book or the details of a particular author

BookDetails
AuthorDetails

Clicking on the hyperlinks for the author on the books detail page or clicking on the hyperlink of the book on the author details page takes us to those pages respectively.

### Feedback
I'd like general feedback on the code and pull request itself, any changes I should add to the code to make sure it follows the style guide. Best practices etc.

### CODE REVIEW

File: catalog/views.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 from .models import Book, Author, BookInstance, Genre
 
+from django.views import generic
+
 # Create your views here.
```
Comment (shahabejaz) [inline catalog/views.py:7]: You can remove this comment, line 5.

File: catalog/views.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
...
 
 def index(request):
```
Comment (shahabejaz) [inline catalog/views.py:9]: Add type hint on this.

**Review by shahabejaz — APPROVED** [review]
Approving this, but before merging, please resolve the comments I have added.

File: catalog/views.py (hunk 1 above)
Comment (RizwanNiazFolio3) [reply catalog/views.py]: Done

File: catalog/views.py (hunk 2 above)
Comment (RizwanNiazFolio3) [reply catalog/views.py]: done

**Pull request merged**